Bug 1770149 - UniFFI: Adjust docs for all metric types (#2071)
mozilla/glean: docs/user/reference/metrics/custom_distribution.md

### `testGetValue`

> Gets the recorded value for a given counter metric.  

Reviewer: ```suggestion
Gets the recorded value for the custom distribution metric.  
```

val snapshot = Graphics.checkerboardPeak.testGetValue()

// Does the sum have the expected value?
assertEquals(23, snapshot.sum)

> // Usually you don't know the exact timing values, but how many should have been recorded.

Reviewer: ```suggestion
// Usually you don't know the exact values, but how many should have been recorded.
```

assertEquals(1L, snapshot.values.size)
```

</div>

<div data-lang="Java" class="tab">

```Java
import org.mozilla.yourApplication.GleanMetrics.Graphics;

// Get snapshot
val snapshot = Graphics.INSTANCE.checkerboardPeak().testGetValue();

// Does the sum have the expected value?
assertEquals(23, snapshot.sum);
```

</div>
<div data-lang="Swift" class="tab">

```Swift
// Get snapshot.
let snapshot = graphics.checkerboardPeak.testGetValue()

> // Usually you don't know the exact timing values,

Reviewer: timing? This is a custom distribution, sir.

Author: coooopy-pasta.

> // but how many should have been recorded.

Reviewer: In which case shouldn't we include an example of checking that?

XCTAssertEqual(23, snapshot.sum)
```

</div>
<div data-lang="Python" class="tab">

```Python
from glean import load_metrics
metrics = load_metrics("metrics.yaml")

# Get snapshot.
snapshot = metrics.graphics.checkerboard_peak.test_get_value()

> # Usually you don't know the exact timing values,

Reviewer: ```suggestion
# Usually you don't know the exact values,
```

{{#include ../../../shared/tab_footer.md}}

### `testGetNumRecordedErrors`

> Gets number of errors recorded for a given counter metric.

Reviewer: ```suggestion
Gets the number of errors recorded for the custom distribution metric.
```

// Were any of the values negative and thus caused an error to be recorded?
assert_eq!(
    0,
    graphics::checkerboard_peak.test_get_num_recorded_errors(
        ErrorType::InvalidValue,

>         None

Reviewer: What's this second param for? counter and boolean don't have it in Rust (that I can see)

Author: haha, see. I made a note in my `notes.txt` to go back and fix that everywhere:
`test_get_num_recorded_errors` takes as the second parameter the (optional) ping to look into because ... well, technically you can look in other pings (and on Rust we can't leave it out).

Now the thing is: we don't actually need that: errors _always_ also go to the `metrics` ping on top of whatever ping this metric normally goes in. So we could be safe just removing that and always assuming `metrics`.
That's a bigger change though, so I'm filling a bug for that.
And because of that I'm less motivated to fix it everywhere just to revert it soon after.

Author: https://bugzilla.mozilla.org/show_bug.cgi?id=1771705
